### PR TITLE
feat: Template Input block refactoring

### PR DESCRIPTION
**Pull request by ruslan-sazonov**
This PR introduces a new `legacyTemplateInput` block type which is technically an old `templateInput` block type.

### CODE REVIEW

File: packages/sidebar-settings/src/blocks/legacyTemplateInput.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import type { BaseBlock } from './base';
+
+export type LegacyTemplateInputBlock<AppBridge> = {
+    /**
+     * The setting type.
+     */
+    type: 'legacyTemplateInput';
+
+    /**
+     * Whether multiple assets can be selected.
```
Comment (SamuelAlev) [inline packages/sidebar-settings/src/blocks/legacyTemplateInput.ts:12]:
```suggestion
     * Whether multiple templates can be selected.
```

Comment (ruslan-sazonov) [reply packages/sidebar-settings/src/blocks/legacyTemplateInput.ts]: Fixed typo

File: packages/sidebar-settings/src/blocks/legacyTemplateInput.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export type LegacyTemplateInputBlock<AppBridge> = {
+    /**
+     * The setting type.
+     */
+    type: 'legacyTemplateInput';
+
+    /**
+     * Whether multiple templates can be selected.
+     */
+    multiSelection?: boolean;
```
Comment (SamuelAlev) [inline packages/sidebar-settings/src/blocks/legacyTemplateInput.ts:14]: `multiSelection` is also defined in `TemplateInputBlock` type but never used

Comment (ruslan-sazonov) [reply packages/sidebar-settings/src/blocks/legacyTemplateInput.ts]: I've removed `multiSelection` for now for both blocks. This prop came from an asset input block which I used as an example for a template block.

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**